include/verilated_vpi.cpp

```diff
@@ -1901,9 +1946,17 @@ vpiHandle vpi_put_value(vpiHandle object, p_vpi_value value_p, p_vpi_time /*time
         _VL_VPI_ERROR(__FILE__, __LINE__, "%s: Unsupported format (%s) as requested for %s",
                       VL_FUNC, VerilatedVpiError::strFromVpiVal(value_p->format), vop->fullname());
         return NULL;
+    } else if (VerilatedVpioParam* vop = VerilatedVpioParam::castp(object)) {
+        _VL_VPI_WARNING(__FILE__, __LINE__, "%s: Ignoring vpi_put_value to vpiParameter: %s",
+                        VL_FUNC, vop->fullname());
+        return 0;
+    } else if (VerilatedVpioConst* vop = VerilatedVpioConst::castp(object)) {
+        _VL_VPI_WARNING(__FILE__, __LINE__, "%s: Ignoring vpi_put_value to vpiConstant: %s",
+                        VL_FUNC, vop->fullname());
+        return 0;
```

Comment:
OK, I've added some more tests to cover the warnings. 
Is there a tool in the repo to track code coverage?

Reply:
You can run nodist/code_coverage and look at the resulting HTML report.  This will take about 8 CPU*hours.  I'm ok with merging without this complete (as then the cron job on master would see it) as long as you agree to fixup any test holes later.